### ISSUE TITLE
Using `Dataset.map` with `n_proc>1` print multiple progress bars

### ISSUE THREAD

**Issue by sgugger** [huggingface/datasets#771]
When using `Dataset.map` with `n_proc > 1`, only one of the processes should print a progress bar (to make the output readable). Right now, `n_proc` progress bars are printed.

**Comment by lhoestq** [comment]
Yes it allows to monitor the speed of each process. Currently each process takes care of one shard of the dataset.

At one point we can consider using streaming batches to a pool of processes instead of sharding the dataset in `num_proc` parts. At that point it will be easy to use only one progress bar

**Comment by prateeky2806** [comment]
Hi @lhoestq, I am facing a similar issue, it is annoying when lots of progress bars are printed. Is there a way to turn off this behavior?
